rpcserver.go

```diff
@@ -7729,9 +7729,132 @@ func (r *rpcServer) getInboundPolicy(ctx context.Context, chanID uint64,
 
 	return policy, nil
 }
+
+// assetInvoiceAmt calculates the amount of asset units to pay for an invoice
+// which is expressed in sats.
+func (r *rpcServer) assetInvoiceAmt(ctx context.Context, assetID asset.ID,
+	invoiceAmt lnwire.MilliSatoshi, peerPubKey *route.Vertex,
+	expiryTimestamp time.Time) (uint64, error) {
+
+	acceptedQuote, err := r.fetchSendRfqQuote(
+		ctx, assetID, invoiceAmt, peerPubKey, expiryTimestamp,
+	)
+	if err != nil {
+		return 0, fmt.Errorf("error sending RFQ quote: %w", err)
+	}
+
+	return acceptedQuote.AssetAmount, nil
+}
+
 // DecodeAssetPayReq decodes an incoming invoice, then uses the RFQ system to
 // map the BTC amount to the amount of asset units for the specified asset ID.
 func (r *rpcServer) DecodeAssetPayReq(ctx context.Context,
 	payReq *tchrpc.AssetPayReqString) (*tchrpc.AssetPayReq, error) {
+
+	// First, we'll perform some basic input validation.
+	switch {
+	case len(payReq.AssetId) == 0:
+		return nil, fmt.Errorf("asset ID must be specified")
+
+	case len(payReq.AssetId) != 32:
+		return nil, fmt.Errorf("asset ID must be 32 bytes, "+
+			"was %d", len(payReq.AssetId))
+
+	case len(payReq.PayReqString.PayReq) == 0:
+		return nil, fmt.Errorf("payment request must be specified")
+	}
+
+	var (
+		resp    tchrpc.AssetPayReq
+		assetID asset.ID
+	)
```

Comment:
Commit message refers to a `DecodeAssetInvoice` but the new method name is `DecodeAssetPayReq`